packages/client/src/module.js

```diff
@@ -30,6 +34,9 @@ const reducer = (state = initialState, action) => {
       return state.set('tokens', Map(tokens));
     }
     case CLEAR_TOKENS: {
+      if (!state.get('isImpersonated')) {
```

Comment:
if not impersonated, clear original tokens? If you are not impersonated there should not be tokens in the first place isn't there? And if you are impersonated, what clearing original token mean?
I think you should remove that from this reducer and handle that on `SET_IMPERSONATED` false.